docs/docs/dev_docs/contracts/deploying.md

```diff
@@ -27,35 +59,101 @@ Set this only if this contract is expected to receive private notes (in such a c
+Because the contract sends a note to the owner specified inside the constructor, we need their public key to encrypt the note with. For this, we first need to register the owner as a recipient inside the Aztec RPC with the following command:
+
+<Tabs groupId="deployment-methods">
+<TabItem value="cli" label="Aztec CLI">
 
 ```bash
 aztec-cli register-recipient --address 0x147392a39e593189902458f4303bc6e0a39128c5a1c1612f76527a162d36d529 --public-key 0x26e193aef4f83c70651485b5526c6d01a36d763223ab24efd1f9ff91b394ac0c20ad99d0ef669dc0dde8d5f5996c63105de8e15c2c87d8260b9e6f02f72af622 --partial-address 0x200e9a6c2d2e8352012e51c6637659713d336405c29386c7c4ac56779ab54fa7
 ```
 
+</TabItem>
+<TabItem value="js" label="Aztec.js">
+
+```ts
+await aztecRpc.registerRecipient(completeAddress);
+```
+
+
+</TabItem>
+</Tabs>
+
+When you create a new account, it gets automatically registered. It can be verified by calling `aztec-cli get-accounts` OR in aztec.js by using `await aztecRpc.getAccounts()`
+
 > **NOTE 1**: If we didn't register owner as a recipient we could not encrypt a note for the owner and the contract deployment would fail because constructor execution would fail (we need owner's public key to encrypt a note).
 
 > **NOTE 2**: If a note recipient is one of the accounts inside the Aztec RPC, we don't need to register it as a recipient because we already have the public key available.
 
 Once the recipient is registered we can deploy the contract:
 
+<Tabs groupId="deployment-methods">
+<TabItem value="cli" label="Aztec CLI">
+
 ```bash
 aztec-cli deploy PrivateTokenContractAbi --args 1000 0x147392a39e593189902458f4303bc6e0a39128c5a1c1612f76527a162d36d529
 ```
 
+</TabItem>
+<TabItem value="js" label="Aztec.js">
+
+```ts
+// PrivateTokenContract is the TS interface that is automatically generated when compiling the contract with the `-ts` flag.
+const initialBalance = 1000n;
+const owner = AztecAddress.from("0x147392a39e593189902458f4303bc6e0a39128c5a1c1612f76527a162d36d529");
+contract = await PrivateTokenContract.deploy(wallet, initialBalance, owner).send().deployed();
+logger(`Contract deployed at ${contract.address}`);
+```
+
+</TabItem>
+</Tabs>
+
 If everything went as expected you should see the following output (with a different address):
 > Contract deployed at 0x151de6120ae6628129ee852c5fc7bcbc8531055f76d4347cdc86003bbea96906
 
 If we pass the salt as an argument:
 
+<Tabs groupId="deployment-methods">
+<TabItem value="cli" label="Aztec CLI">
+
 ```bash
 aztec-cli deploy PrivateTokenContractAbi --args 1000 0x147392a39e593189902458f4303bc6e0a39128c5a1c1612f76527a162d36d529 --salt 0x123
 ```
 
+</TabItem>
+<TabItem value="js" label="Aztec.js">
+
+```ts
+contract = await PrivateTokenContract.deploy(wallet, initialBalance, owner).send({ contractAddressSalt: Fr.fromString("0x123") }).deployed();
```

Comment:
More api wise, but why are we not just calling it a salt?

Reply:
unsure 🤷